components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/internal/FrameworkServiceComponent.java

```diff
@@ -477,8 +478,16 @@ protected void unsetRealmService(RealmService realmService) {
             policy = ReferencePolicy.DYNAMIC,
             unbind = "unsetAuthenticator"
     )
-    protected void setAuthenticator(ApplicationAuthenticator authenticator) {
-
+    protected void setAuthenticator(ApplicationAuthenticator authenticator)
+            throws IdentityProviderManagementServerException {
+
+        /* All custom authenticator names must start with the `custom_` prefix. If a system-defined authenticator is
+         attempted to be registered at server startup with a name starting with this prefix, an error will be thrown.*/
+        if (authenticator.getName().startsWith("custom_")) {
```

Comment:
Isn't this hyphen not underscore ?
Shall we have a string constant defined for this that can be used across management layer and in here